### DISCUSSION TITLE
UseTransition pending state

### DISCUSSION THREAD

**Discussion by ScreamZ** [next-safe-action/next-safe-action#218] [Q&A] [answered]
Hi @TheEdoRan 

https://github.com/TheEdoRan/next-safe-action/blob/8207452f479b07cb49ed1ca77a0de5e0d76297cc/packages/next-safe-action/src/hooks.ts#L33

Why do you skip the first pending state of useTransition ?

When using `signIn` method from authJS inside a server action:
- Using `useAction` from next-safe-action, `isExecuting` returns false, before the redirection to the auth page occurs, resulting in a glitchy UI.
- Without using next safe action, and using `const [isPending, setTransition] = useTransition()` and using a server action with `signIn` `isPending` is not `false` before the redirection occurs, which results in nice UI.

I'm waiting for your answer.

Thanks

PS: Maybe `isExecuting` could be: `isExecuting` (internal)  && `isPending` (from `useTransition`) ?

**Comment by TheEdoRan** [comment, marked as answer]
Closing in favor of #232
